docs/rechunking.ipynb

```diff
@@ -0,0 +1,418 @@
+      "WARNING:root:Make sure that locally built Python SDK docker image has Python 3.9 interpreter.\n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage0/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage0/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage2/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage2/Consolidate/GroupByTempKeys'. \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "[(Key(offsets={'x': 0, 'y': 0}, vars=None),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 5, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 0 1 2 3 4 5 6 7 8 9\n",
+       "      bar      (x, y) int64 100 101 102 103 104 105 106 107 108 109),\n",
+       " (Key(offsets={'x': 5, 'y': 0}, vars=None),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 3, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 10 11 12 13 14 15\n",
+       "      bar      (x, y) int64 110 111 112 113 114 115)]"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "inputs | xbeam.Rechunk(dim_sizes={'x': 6}, source_chunks={'x': 3}, target_chunks={'x': 5}, itemsize=8)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`Rechunk` requires specifying a few more parameters, but based on that information it can be _much_ more efficient for more complex rechunking tasks, particular in cases where data needs to be distributed into a very different shape (e.g., distributing a matrix across rows vs. columns). A naive \"splitting\" approach in such cases could divide datasets into extremely small tasks corresponding to individual array elements, which adds a huge amount of overhead."
+   ]
+  }
+ ],
+ "metadata": {
+  "interpreter": {
+   "hash": "aef148d7ea0dbd1f91630322dd5bc9e24a2135d95f24fe1a9dab9696856be2b9"
+  },
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.9.6"
```

Comment:
Interesting -- it seems like you're on the right version -- any idea why you get the warning?